src/components/implementation/no_interface/vkernel/micro_booter.c

```diff
@@ -17,25 +17,31 @@ int vmid;
 void
 vm_init(void *d)
 {
-	vmid = (int)d;
+	int         rcvd, blocked;
+	cycles_t    cycles;
+	thdid_t     tid;
+	tcap_time_t timeout = 0, thd_timeout;
+
+	vmid = cos_sinv(VM_CAPTBL_SELF_SINV_BASE, VK_SERV_VM_ID << 16 | cos_thdid(), 0, 0, 0);
+
 	cos_meminfo_init(&booter_info.mi, BOOT_MEM_KM_BASE, VM_UNTYPED_SIZE, BOOT_CAPTBL_SELF_UNTYPED_PT);
 	cos_compinfo_init(&booter_info, BOOT_CAPTBL_SELF_PT, BOOT_CAPTBL_SELF_CT, BOOT_CAPTBL_SELF_COMP,
-			  (vaddr_t)cos_get_heap_ptr(), vmid == 0 ? DOM0_CAPTBL_FREE : VM_CAPTBL_FREE, &booter_info);
+			(vaddr_t)cos_get_heap_ptr(), vmid == 0 ? DOM0_CAPTBL_FREE : VM_CAPTBL_FREE, &booter_info);
```

Comment:
I'm confused when you say `library` and you've said that so many times in this review. :( 
I don't really have a library, it's only parts of `vkernel` component that are modularised into `vk_api` and parts of it is specific to how the vkernel, vms, dom0 interract with each other in the current vkernel component implementation.

I may be able to make parts of it generic, like, really not call it `vm`, make it a `fork_comp` kind of API. None of the current `vkernel` has that. I totally understand that it is good to make parts of it a generic library but for the current `vkernel`, we don't have a library, that's my confusion when you say `library`. I'll add an issue to possible make a generic library out of it and work on it later. 

Coming to this `micro_booter.c` and `vm_init`, it is the main function of a `vm` and is doing `cos_compinfo_init` for it's component so it starts its `cap frontier` where it knows it should which is why when DOM0 boots up it's frontier is set to a different value than when another VM boots up.